File: in_Java/0098 Validate Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isValidBST(TreeNode root) {
        // inorder
        List<Integer> L = new ArrayList<>();
        if(root == null) return true;
        Stack<TreeNode> st = new Stack<>();
        TreeNode cur = root;
        while(cur != null || !st.isEmpty()){
            while(cur != null){
                st.push(cur);
                cur = cur.left;
            }
            cur = st.pop();
            L.add(cur.val);
            cur = cur.right;
        }
        // check if it is acsending
        for(int i = 1; i < L.size(); i++){
            if(L.get(i) <= L.get(i-1)) return false;
        }
        return true;
    }
}

File: in_Python/0098 Validate Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isValidBST(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root == None:
            return True
        st = []
        res = []
        cur = root
        st.append(cur)
        while len(st) != 0:
            while cur != None:
                st.append(cur)
                cur = cur.left
            n = st.pop()
            res.append(n.val)
            cur = n.right
        for i in range(1, len(res)-1):
            if res[i] <= res[i-1]:
                return False
        return True
            